### PR TITLE
Update to zict 3.0

### PR DESCRIPTION
**Pull request by pentschev**
With the release of zict 3.0 a few changes were made to resources that were used in spilling tests that are being updated here.

### CODE REVIEW

File: dask_cuda/tests/test_spill.py [hunk 1]
```diff
@@ -31,7 +30,8 @@ def device_host_file_size_matches(
     # `dhf.disk` is only available when Worker's `memory_limit != 0`
     if dhf.disk is not None:
         file_path = [
-            os.path.join(dhf.disk.directory, safe_key(k)) for k in dhf.disk.keys()
+            os.path.join(dhf.disk.directory, dhf.disk._safe_key(k))
+            for k in dhf.disk.keys()
```
Comment (crusaderky) [inline dask_cuda/tests/test_spill.py:34]:
```suggestion
            os.path.join(dhf.disk.directory, fname)
            for fname in dhf.disk.filenames.values()
```

Comment (pentschev) [reply dask_cuda/tests/test_spill.py]: Checked this locally and the test pass with that change. Thanks @crusaderky !

**Comment by quasiben** [comment]
rerun tests

**Comment by pentschev** [comment]
rerun tests

**Comment by jakirkham** [comment]
Does [this requirement]( https://github.com/rapidsai/dask-cuda/blob/eed3eb6b02951ac2311b81f546b9053184254c4b/dependencies.yaml#L104 ) need updating?

**Comment by pentschev** [comment]
> Does [this requirement](https://github.com/rapidsai/dask-cuda/blob/eed3eb6b02951ac2311b81f546b9053184254c4b/dependencies.yaml#L104) need updating?

Given the only change required here is to a test, I think it's safer if we keep the pinning somewhat loose. I upgraded it to 2.0 so it doesn't stay "too loose".

**Comment by quasiben** [comment]
rerun tests

**Comment by quasiben** [comment]
Not sure why this was failing before.  Anything else needed here @pentschev  ?

**Comment by pentschev** [comment]
> Not sure why this was failing before.

It was failing before zict 3.0 was release, but it's been passing since. 

> Anything else needed here @pentschev ?

I think we are good to merge.

**Pull request merged**